presto-main/src/main/java/com/facebook/presto/metadata/CastType.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.spi.function.OperatorType;
+
+import static com.facebook.presto.metadata.OperatorSignatureUtils.mangleOperatorName;
+import static com.facebook.presto.operator.scalar.JsonStringToArrayCast.JSON_STRING_TO_ARRAY_NAME;
+import static com.facebook.presto.operator.scalar.JsonStringToMapCast.JSON_STRING_TO_MAP_NAME;
+import static com.facebook.presto.operator.scalar.JsonStringToRowCast.JSON_STRING_TO_ROW_NAME;
+import static com.facebook.presto.operator.scalar.TryCastFunction.TRY_CAST_NAME;
+import static java.lang.String.format;
+
+public enum CastType
+{
+    CAST(mangleOperatorName(OperatorType.CAST.name()), true),
+    SATURATED_FLOOR_CAST(mangleOperatorName(OperatorType.SATURATED_FLOOR_CAST.name()), true),
+    TRY_CAST(TRY_CAST_NAME, false),
+    JSON_TO_ARRAY_CAST(JSON_STRING_TO_ARRAY_NAME, false),
+    JSON_TO_MAP_CAST(JSON_STRING_TO_MAP_NAME, false),
+    JSON_TO_ROW_CAST(JSON_STRING_TO_ROW_NAME, false);
+
+    private final String castName;
```

Comment:
But not all cast are functions... I also lost some sleep on this one. Some cast are operators, others are functions. I gave up. At this point I'm willing to change all CAST to function and remove them from operator. But I don't know all the implications of it. It might not be the safest thing to do.